ci/dhis2-verify-app.yml

```diff
@@ -0,0 +1,164 @@
+name: 'dhis2: verify (app)'
+
+on:
+    push:
+        branches:
+            # match branches in:
+            # https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches
+            - master
+            - next
+            - next-major
+            - alpha
+            - beta
+            - '[0-9]+.x'
+            - '[0-9]+.x.x'
+            - '[0-9]+.[0-9]+.x'
+
+    pull_request:
+
+env:
+    GIT_AUTHOR_NAME: '@dhis2-bot'
+    GIT_AUTHOR_EMAIL: 'apps@dhis2.org'
+    GIT_COMMITTER_NAME: '@dhis2-bot'
+    GIT_COMMITTER_EMAIL: 'apps@dhis2.org'
+    NPM_TOKEN: ${{secrets.DHIS2_BOT_NPM_TOKEN}}
+    GH_TOKEN: ${{secrets.DHIS2_BOT_GITHUB_TOKEN}}
+    CI: true
+
+jobs:
+    install:
+        runs-on: ubuntu-latest
+        if: "!contains(github.event.head_commit.message, '[skip ci]')"
+
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12.x
+
+            - uses: actions/cache@v2
+              id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+              with:
+                  path: '**/node_modules'
+                  key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+
+            - name: Install
+              if: steps.yarn-cache.outputs.cache-hit != 'true'
+              run: yarn install --frozen-lockfile
+
+    build:
+        runs-on: ubuntu-latest
+        needs: install
+        if: "!contains(github.event.head_commit.message, '[skip ci]')"
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12.x
+
+            - uses: actions/cache@v2
+              id: yarn-cache
+              with:
+                  path: '**/node_modules'
+                  key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+
+            - name: Install
+              if: steps.yarn-cache.outputs.cache-hit != 'true'
+              run: yarn install --frozen-lockfile
```

Comment:
I agree that it shouldn't be necessary. I had some problems related to this at first, but I think that after iterating a few times on these it should work without it. Let me try and remove it and see what happens in e.g. UI.

Reply:
Seems like it works now that we restore the entire node_modules and install the cypress binary explicitly:

https://github.com/dhis2/ui/runs/1273953364

Reply:
972c1a1

Reply:
Nice!  So then we should remove the other `yarn install` steps?

For Cypress we should probably use the [official Cypress action](https://github.com/cypress-io/github-action) (which installs the binary) and should also definitely be running a matrix of Cypress jobs in parallel so that it doesn't take 12+ minutes (that run is still going), but that's probably outside the scope of this change (we should have a separate e2e workflow maybe...?  Or add it to the same one later?

Reply:
(I see you did this already :-) )